xls/dslx/bytecode/bytecode_interpreter_test.cc

```diff
@@ -1519,6 +1519,42 @@ fn main() {
   EXPECT_TRUE(value.IsUnit());
 }
 
+// https://github.com/google/xls/issues/981
+TEST(BytecodeInterpreterTest, AssertEqFailProcIterations) {
+  constexpr std::string_view kProgram = R"(
+#[test_proc]
+proc BTester {
+    terminator: chan<bool> out;
+
+    init { (u32:0) }
+
+    config(terminator: chan<bool> out) {
+        (terminator,)
+    }
+
+    next(tok: token, state: u32) {
+        assert_eq(state, u32:0);
+        // Assert_eq type deduction didn't fail in
```

Comment:
would suggest a slight rephrasing to be shorter:
```
ensure at least 2 `next()` iterations to create an interpreter frame.
```

Reply:
Fixed.